core/include/gnuradio-4.0/TriggerMatcher.hpp

```diff
@@ -0,0 +1,302 @@
+#ifndef GNURADIO_TRIGGERMATCHER_HPP
+#define GNURADIO_TRIGGERMATCHER_HPP
+
+#include "gnuradio-4.0/Message.hpp"
+#include "gnuradio-4.0/meta/formatter.hpp"
+#include "gnuradio-4.0/Tag.hpp"
+
+namespace gr::trigger {
+
+constexpr inline std::string_view SEPARATOR = "/";
+constexpr inline std::string_view RANGE_OP  = ",";
+
+enum class MatchResult {
+    Matching,    ///< Start a new dataset
+    NotMatching, ///< Finish dataset
```

Comment:
Just a suggestion: how about renaming it to Start and Finish, or something similar? NotMatching seems to imply ignoring the tag if no match is found.

Reply:
Yes, correct... the commented documentation is one of the use-cases but we need the `NotMatching` case also when there is not 'start->stop' relationship.